tests/Mcfedr/PeriodicQueueDriverBundle/Tests/Manager/PeriodicQueueManagerTest.php

```diff
@@ -45,39 +46,60 @@ public function setUp()
-                    return false;
-                }
-                if (!isset($options['time']) || !$options['time'] instanceof \DateTime) {
-                    return false;
-                }
-
-                return true;
-            }), 'delay')
+            ->with('mcfedr_periodic_queue_driver.worker',
+                $this->callback(function ($arguments) use ($pattern) {
+                    $this->assertCount(6, $arguments);
+                    $this->assertArrayHasKey('name', $arguments);
+                    $this->assertEquals('test_worker', $arguments['name']);
+                    $this->assertArrayHasKey('arguments', $arguments);
+                    $this->assertCount(1, $arguments['arguments']);
+
+                    $this->assertArrayHasKey('argument_a', $arguments['arguments']);
+                    $this->assertEquals('a', $arguments['arguments']['argument_a']);
+
+                    $this->assertArrayHasKey('job_tokens', $arguments);
+                    $this->assertCount(2, $arguments['job_tokens']);
+                    $this->assertArrayHasKey('token', $arguments['job_tokens']);
+                    $this->assertRegExp($pattern, $arguments['job_tokens']['token']);
+                    $this->assertArrayHasKey('next_token', $arguments['job_tokens']);
+                    $this->assertRegExp($pattern, $arguments['job_tokens']['next_token']);
+
+                    $this->assertArrayHasKey('period', $arguments);
+                    $this->assertEquals(3600, $arguments['period']);
+                    $this->assertArrayHasKey('delay_options', $arguments);
+                    $this->assertCount(1, $arguments['delay_options']);
+                    $this->assertArrayHasKey('delay_manager_option_a', $arguments['delay_options']);
+                    $this->assertEquals('a', $arguments['delay_options']['delay_manager_option_a']);
+                    $this->assertArrayHasKey('delay_manager', $arguments);
+                    $this->assertEquals('delay', $arguments['delay_manager']);
+
+                    return true;
+                }),
+                $this->callback(function ($options) {
+                    if (!is_array($options)) {
+                        return false;
+                    }
+                    if (!isset($options['delay_manager_option_a']) || $options['delay_manager_option_a'] != 'a') {
+                        return false;
+                    }
+                    if (!isset($options['time']) || !$options['time'] instanceof \DateTime) {
+                        return false;
+                    }
+
+                    return true;
+                }), 'delay')
             ->willReturn($fakeJob);
 
         $job = $this->manager->put('test_worker', [
             'argument_a' => 'a',
         ], ['period' => 3600]);
 
-        $this->assertEquals($fakeJob, $job);
+        $this->assertInstanceOf(PeriodicJob::class, $job);
+        $this->assertRegExp($pattern, $job->getJobToken());
```

Comment:
I think we should not check the pattern, but just check that it is a non empty string